scripts/format_notebooks.py

```diff
@@ -181,6 +181,58 @@ def preprocess(self, nb, resources):
         return nb, resources
 
 
+class VersionCheckPreprocessor(preprocessors.Preprocessor):
+    metadata_tag = "VersionCheck"  # tag for added cells so that we can find them easily
+
+    try:
+        # determine the current stellargraph version by importing the library's version file
+        sys.path.append(
+            (
+                os.path.abspath(
+                    os.path.join(
+                        os.path.dirname(__file__), os.path.pardir, "stellargraph"
+                    )
+                )
+            )
+        )
+        import version as sgversion
+
+        stellargraph_version = sgversion.__version__
+    except ImportError as error:
+        print(f"ERROR: unable to import StellarGraph version: {error.message}")
+        stellargraph_version = None
+
+    version_check_code = f"""\
+# verify that we're using the correct version of StellarGraph for this notebook
+import stellargraph as sg
+from packaging import version
+
+if version.parse(sg.__version__) < version.parse("{stellargraph_version}"):
+    raise ValueError(
+        f"The installed version {{sg.__version__}} of StellarGraph is older than the version {stellargraph_version} required for this notebook. Please update StellarGraph, eg: pip install --upgrade stellargraph"
```

Comment:
A few minor things style things:

- e.g. (and i.e.) can be confusing, so it's often recommended to just spell them out (for instance, https://docs.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/e/eg)
- delimiting the `pip` invocation makes it a bit clearer where it starts and ends.

```suggestion
        f"The installed version {{sg.__version__}} of StellarGraph is older than the version {stellargraph_version} required for this notebook. Please update StellarGraph, for instance: 'pip install --upgrade stellargraph'"
```